### ISSUE TITLE
Enterprise Hosted Gitlab Call invoked without "/api/v4" causing 404 Error leading to "wrong project_slug or project_id"

### ISSUE THREAD

**Issue by smaadavan** [immobiliare/backstage-plugin-gitlab#381]
After integration, when tried to load Gitlab entity, it fails as shown in screenshot.  

![image](https://github.com/immobiliare/backstage-plugin-gitlab/assets/83218559/742e5ee9-7677-4e6a-b775-e7586c4f772e)

Upon checking in browser console network tab, could see URL as

`http://{BACKEND_FQDN}:7007/api/gitlab/rest/{ENTERPRISE_GITLAB_DNS}/projects/{GROUP_PATH_AND_PROJECT_PATH_URL_ENCODED} `
Above URL resulting in 404 error.

Ideally, it should either be without projects or with "/api/v4" 

`http://{BACKEND_FQDN}:7007/api/gitlab/rest/{ENTERPRISE_GITLAB_DNS}/{GROUP_PATH_AND_PROJECT_PATH_URL_ENCODED}
`
or

`http://{BACKEND_FQDN}:7007/api/gitlab/rest/{ENTERPRISE_GITLAB_DNS}/api/v4/projects/{GROUP_PATH_AND_PROJECT_PATH_URL_ENCODED}`

Have set gitlab integrations properly in app-config.yaml as below

```
gitlab:
    - host: {ENTERPRISE_GITLAB_DNS}
      token: XYZ
      apiBaseUrl: https://{ENTERPRISE_GITLAB_DNS}/api/v4/

```

Tried adding annotations as well in component section yaml file as below.

```
annotations:
    gitlab.com/project-slug: {GROUP}/{PROJECT_NAME}
    gitlab.com/project-id: '12345' (sample value)
```

**Comment by antoniomuso** [comment]
The URLs are correct. Have you followed the [setup guide](https://github.com/immobiliare/backstage-plugin-gitlab?tab=readme-ov-file#setup) in all the steps? You haven't configured the backend plugin.

**Comment by smaadavan** [comment]
Thanks @antoniomuso.

For others who are facing similar issue:

As directed, tried verifying backend plugin. My existing backend system was old way, but for Gitlab backend plugin, I tried installing new backend system packages & registered for new backend system hoping it would work in conjunction to exiting plugins using old backend system. 

Probably it does not work that way or I implemented wrongly for both old & new backend system to work concurrently. Reverted those changes & used "useHotMemoize" itself with "wrong project_slug or project_id" error resolved and got "Unable to fetch" error.

For "Unable to fetch" error, it was browser restriction not allowing with http protocol to jump between hosts. When I configured a certificate for the baseUrl, it started working fine. Thanks.
